src/dotnet-install.sh

```diff
@@ -546,6 +546,36 @@ is_dotnet_package_installed() {
     fi
 }
 
+# args:
+# downloaded file - $1
+# remote_file_size - $2
+validate_remote_local_file_sizes() 
+{
+    eval $invocation
+
+    local downloaded_file="$1"
+    local remote_file_size="$2"
+
+    local file_size=''
+    if [[ "$OSTYPE" == "linux-gnu"* ]]; then
+        file_size="$(stat -c '%s' "$downloaded_file")"
+    elif [[ "$OSTYPE" == "darwin"* ]]; then
+        file_size="$(stat -f '%z' "$downloaded_file")"
+    fi  
+    
+    if [ -n "$file_size" ]; then
+        say "Downloaded file size is $file_size bytes."
+
+        if [ -n "$remote_file_size" ] && [ "$file_size" != "$remote_file_size" ]; then
+            say "The remote and local file sizes are not equal. Remote file size is $remote_file_size bytes and local size is $file_size bytes. The local package may be corrupted."
+        else
+            say "The remote and local file sizes are equal."
```

Comment:
The way the conditions are written now, we print this message if we were not able to get the remote size. This looks unexpected.